cmd/catalyst-node/catalyst-node.go

```diff
@@ -424,18 +424,58 @@ func writeSerfConfig(config *agent.Config) (string, error) {
 }
 
 func startCatalystWebServer(httpAddr string, redirectPrefixes []string) {
-	http.Handle("/hls/", redirectHlsHandler(redirectPrefixes))
+	http.Handle("/", redirectHandler(redirectPrefixes))
 	glog.Infof("HTTP server listening on %s", httpAddr)
 	glog.Fatal(http.ListenAndServe(httpAddr, nil))
 }
 
 var getClosestNode = queryMistForClosestNode
 
-func redirectHlsHandler(redirectPrefixes []string) http.Handler {
+// return the best node available for a given stream. will return any node if nobody has the stream.
+func getBestNode(redirectPrefixes []string, playbackID, lat, lon string) (string, string, error) {
+	var nodeAddr, fullPlaybackID, fallbackAddr string
+	var err error
+	var waitGroup sync.WaitGroup
+
+	for _, prefix := range redirectPrefixes {
+		waitGroup.Add(1)
+		go func(prefix string) {
+			addr, e := getClosestNode(playbackID, lat, lon, prefix)
+			if e != nil {
+				err := e
```

Comment:
This line is wrong. I guess what you wanted to have is `err = e`. Otherwise, you just create a new variable, which is not used anywhere.

On the other hand if you use `err = e`, then you need to think how to sync it with the other goroutine.

Reply:
Good catch, wasn't really sure how to handle this case. I guess I was thinking that I didn't really care which error that I returned from the outer error handler; presumably the requests failed for the same reason and we could just return any of them. So maybe I don't care about the sync. But it's not very elegant for sure.

Reply:
I think that in Go, you MUST care about the sync. Because the assignment operation itself is not atomic. So, in the worse case scenario, you may end up here with some panic / seg fault.

Reply:
Oooh, I didn't know that. That'd be a problem even before my change then, yeah? Both goroutines might concurrently modify `nodeAddr`... do you have an idea for a better pattern here? I want something like `Promise.all` in JavaScript where I can get a list of the errors/results afterward, I think...

Reply:
Yeah, I think you're right. That the problem was there even before you changes. You can just use a mutex to solve it.

Other options are channels or some library to have atomic string, but I think it's an overkill for your simple scenario.

Reply:
Added a mutex